libr/core/io.c

```diff
@@ -97,23 +97,42 @@ R_API int r_core_dump(RCore *core, const char *file, ut64 addr, ut64 size, int a
 
 R_API int r_core_write_op(RCore *core, const char *arg, char op) {
 	int i, j, len, ret = false;
-	char *str;
+	char *str = NULL;
 	ut8 *buf;
 
 	// XXX we can work with config.block instead of dupping it
 	buf = (ut8 *)malloc (core->blocksize);
-	str = (char *)malloc (strlen (arg)+1);
-	if (buf == NULL || str == NULL)
+	if (!buf)
 		goto beach;
 	memcpy (buf, core->block, core->blocksize);
-	if (op!='e') {
-		len = r_hex_str2bin (arg, (ut8 *)str);
-		if (len < 0) {
-			eprintf ("Invalid hexpair string\n");
+
+	// fill key buffer either from arg or from clipboard
+	if (arg) {  // parse arg for key
+		str = (char *)malloc (strlen (arg)+1);
+		if (!str)
 			goto beach;
-		}
-	} else len = 0;
+		if (op!='e') {
+			len = r_hex_str2bin (arg, (ut8 *)str);
+			if (len < 0) {
+				eprintf ("Invalid hexpair string\n");
+				goto beach;
+			}
+		} else len = 0;
+	} else {  // use clipboard as key
+		if (op!='e') {
+			len = core->yank_buf->length;
+			if (len <= 0) {
+				eprintf ("Clipboard is empty\n");
+				goto beach;
+			}
+			str = (char *)malloc (len);
```

Comment:
you can use `str = r_mem_dup(core->yank_buf->buf, len);` to save some lines here